src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -92,16 +102,88 @@ protected override Payload ConstructHeadersPayload(BlockLocator locator, uint256
         }
+
+        /// <summary>
+        /// Determines whether the specified peer supports Proven Headers.
+        /// </summary>
+        /// <param name="peer">The peer.</param>
+        /// <returns>
+        ///   <c>true</c> if is peer is PH enabled; otherwise, <c>false</c>.
+        /// </returns>
+        private bool IsPeerPhEnabled(INetworkPeer peer)
+        {
+            return peer.Version >= NBitcoin.Protocol.ProtocolVersion.PROVEN_HEADER_VERSION;
+        }
+
+        /// <summary>
+        /// Determines whether the specified peer is Whitelisted.
+        /// </summary>
+        /// <param name="peer">The peer.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified peer is Whitelisted; otherwise, <c>false</c>.
+        /// </returns>
+        private bool IsPeerWhitelisted(INetworkPeer peer)
+        {
+            return peer.Behavior<IConnectionManagerBehavior>()?.Whitelisted == true;
+        }
+
+        /// <summary>
+        /// Builds the GetHeadersPayload.
+        /// </summary>
+        /// <returns>The GetHeadersPayload instance.
+        /// If the peer can serve PH, GetProvenHeadersPayload is returned, otherwise if it's a legacy peer but it's whitelisted,
+        /// GetHeadersPayload is returned.
+        /// If the attached peer is a legacy peer and it's not whitelisted, returns null.
+        /// </returns>
+        protected override GetHeadersPayload BuildGetHeadersPayload()
+        {
+            INetworkPeer peer = this.AttachedPeer;
+
+            if (IsPeerPhEnabled(peer))
+            {
+                return new GetProvenHeadersPayload()
+                {
+                    BlockLocator = (this.ExpectedPeerTip ?? this.consensusManager.Tip).GetLocator(),
+                    HashStop = null
+                };
+            }
+            // If the peer doesn't supports PH but it's whitelisted, issue a standard GetHeadersPayload
+            else if (IsPeerWhitelisted(peer))
+            {
+                return new GetHeadersPayload()
+                {
+                    BlockLocator = (this.ExpectedPeerTip ?? this.consensusManager.Tip).GetLocator(),
+                    HashStop = null
+                };
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Processes the legacy GetHeaders message.
+        /// Only whitelisted legacy peers are allowed to handle this message.
+        /// </summary>
+        /// <param name="peer">The peer.</param>
+        /// <param name="headers">The headers.</param>
+        /// <returns></returns>
+        protected Task ProcessLegacyHeadersAsync(INetworkPeer peer, List<BlockHeader> headers)
+        {
+            // Only whitelisted legacy peers are allowed to handle this message.
+            if (!IsPeerPhEnabled(peer) && IsPeerWhitelisted(peer))
```

Comment:
Very nice so basically, if we get a normal header (without the proofs its not fake) and PH is not activated or the peer is white listed we allow to processes it.